static/app/components/modals/inviteMembersModal/inviteMembersFooter.spec.tsx

```diff
@@ -0,0 +1,140 @@
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {
+  defaultInviteProps,
+  InviteMembersContext,
+} from 'sentry/components/modals/inviteMembersModal/inviteMembersContext';
+import InviteMembersFooter from 'sentry/components/modals/inviteMembersModal/inviteMembersFooter';
+
+describe('InviteRowControlNew', function () {
+  const getComponent = props => (
+    <InviteMembersContext.Provider value={props}>
+      <InviteMembersFooter canSend />
+    </InviteMembersContext.Provider>
+  );
+
+  it('disables send button when there are no emails', function () {
+    render(getComponent(defaultInviteProps));
+
+    const sendButton = screen.getByLabelText(/send invite/i);
+    expect(sendButton).toBeDisabled();
+  });
+
+  it('enables send button when there are emails', async function () {
+    const mockSendInvites = jest.fn();
+    render(
+      getComponent({
+        ...defaultInviteProps,
+        invites: [
+          {
+            email: 'moo-deng@email.com',
+            role: 'member',
+            teams: new Set<string>(['moo-deng']),
+          },
+        ],
+        sendInvites: mockSendInvites,
+      })
+    );
+
+    const sendButton = screen.getByLabelText(/send invite/i);
+    expect(sendButton).toBeEnabled();
+    await userEvent.click(sendButton);
+    expect(mockSendInvites).toHaveBeenCalled();
+  });
+
+  it('displays correct status message for 1 sent invite', function () {
+    render(
+      getComponent({
+        ...defaultInviteProps,
+        complete: true,
+        inviteStatus: {
+          'moo-deng': {sent: true},
+        },
+        willInvite: true,
+      })
+    );
+    expect(screen.getByText(/sent/i)).toBeInTheDocument();
+    expect(screen.getByText(/1 invite/i)).toBeInTheDocument();
+  });
+
+  it('displays correct status message for multiple sent invites', function () {
+    render(
+      getComponent({
+        ...defaultInviteProps,
+        complete: true,
+        inviteStatus: {
+          'moo-deng': {sent: true},
+          'moo-waan': {sent: true},
+        },
+        willInvite: true,
+      })
+    );
+    expect(screen.getByText(/sent/i)).toBeInTheDocument();
+    expect(screen.getByText(/2 invites/i)).toBeInTheDocument();
+  });
+
+  it('displays correct status message for failed invites', function () {
+    render(
+      getComponent({
+        ...defaultInviteProps,
+        complete: true,
+        inviteStatus: {
+          'moo-deng': {sent: false, error: 'Error'},
+          'moo-waan': {sent: false, error: 'Error'},
+        },
+        willInvite: true,
+      })
+    );
+    expect(screen.getByText(/sent , 2 failed to send\./i)).toBeInTheDocument();
+    expect(screen.getByText(/0 invites/i)).toBeInTheDocument();
```

Comment:
a little janky but had to test it this way because the sent invite count is in a separate `<strong>` component 🫠

Reply:
hm - may just want to target that tag and just check the number in that case? i tend to not like targeting tests with `text` as that's not as hardcoded as you'd always think (like if we wanted to run tests in multiple languages to ensure everything is translated, these would fail). i normally add a `data-` attribute to be really specific with my targeting, if i can't use the other selectors to find something.

Reply:
update: fixed this for successfully sent invites, but still using the text "failed" to look for the failed invite count. i plan on changing the wording of the status message in my next PR, so this will get fixed 👍